rust/tw_keypair/src/tw/mod.rs

```diff
@@ -0,0 +1,64 @@
+// Copyright © 2017-2023 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+mod private;
+mod public;
+
+pub use private::TWPrivateKey;
+pub use public::TWPublicKey;
+
+#[repr(C)]
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum TWCurve {
```

Comment:
`TW` prefix is a C convention because there is no namespace in C, if we implement it in Rust, no need to add `TW` prefix I guess?

Reply:
Sure we can rename. @Milerius what do you think?

Reply:
> Sure we can rename. @Milerius what do you think?

Why? It's actually a struct that is generated in C through CBindgen, the idea is too have our API generated from Rust to C at some point, make sense to keep the same FFI name.

Reply:
We can add a wrapper `struct TWPrivateKey(tw::PrivateKey);` and export it through FFI

Reply:
Later, when we have our own c-bindgen tool, we could do the following:
```rust
use crate::tw::PrivateKey;

#[derive(TWExport)]
pub struct TWPrivateKey(PrivateKey);

impl TWPrivateKey {
  #[derive(TWExportStatic, rename = "TWPrivateKeyCreateWithData")]
  pub fn new(data: Vec<u8>) -> Ptr<TWPrivateKey> { todo!() }
}
```

Reply:
This file is not generated tho, I like @satoshiotomakan's idea, no need to keep same name and should be natural to the implementation language Rusty style.

Later you can generate whatever C files you want